### PR TITLE
lsp.py enhancements

### PR DESCRIPTION
**Pull request by Marenz**
* Add support for sub directories
* Fail fataly when importing outside the test directories

### CODE REVIEW

File: test/lsp.py [hunk 1]
```diff
@@ -537,8 +541,7 @@ def test_diagnostics(self):
                         marker=markers[expected_diagnostic.marker]
                     )
 
-        except Exception as e:
-            print(e)
+        except Exception:
```
Comment (christianparpart) [inline test/lsp.py:589]: I know in the review of the other PR you verbally argued explicitly for keeping the line you just removed now.
What was that reason again and why is it removed now? :)

Comment (Marenz) [reply test/lsp.py:589]: The reason was that another place in the code printed it properly ^^ I guess it was just an extra debug output at the time :innocent:

**Review by christianparpart — APPROVED** [review]

**Review by Sticknmove — COMMENTED** [review]
Thanks

**Comment by Sticknmove** [comment]
Thanks

**Comment by Sticknmove** [comment]
Thanks

**Comment by Sticknmove** [comment]
Thanks

**Pull request merged**